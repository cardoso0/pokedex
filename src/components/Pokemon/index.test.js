import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"

import { Pokemon } from "."

const pokemon = {
  name: "Pikachu",
  sprites: "front_default",
  id: 1,
  types: [{type: {name: "electric"}}],
}

const renderPokemon = 
<Pokemon pokemon={pokemon} large="string" medium="string"/>

describe("Pokemon Component", () => {
  it ("component rendering | snapshot", () => {
    const tree = renderer
    .create(renderPokemon)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ("component rendering", () => {
    expect(render(renderPokemon)).toBeTruthy
  })
})

describe("Rendering pokemon data", () => {
  it ("render pokemon name", () => {
    render(renderPokemon)
    const name = screen.getByText("Pikachu")
    expect(name).toBeTruthy()
  })

  it ("render pokemon id", () => {
    render(renderPokemon)
    const id = screen.getByText("#1")
    expect(id).toBeTruthy()
  })

  it("checking img alt", () => {
    render(renderPokemon)
    const img = screen.getByRole("img")
    expect(img).toHaveAttribute("alt", "Pikachu")
  })

  it("render pokemon type", () => {
    render(renderPokemon)
    const type = screen.getByText("Electric")
    expect(type).toBeTruthy()
  })

})