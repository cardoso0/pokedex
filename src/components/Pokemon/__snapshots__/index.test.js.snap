// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Component component rendering | snapshot 1`] = `
<div
  className="sc-bczRLJ jbmRuO"
>
  <div
    className="sc-gsnTZi eXcMNu"
  >
    <div
      className="pokemonId"
    >
      #
      1
    </div>
    <div
      className="pokemonName"
    >
      Pikachu
    </div>
    <div
      className="heart"
    />
  </div>
  <div
    className="img"
  >
    <img
      alt="Pikachu"
      src="https://cdn.traction.one/pokedex/pokemon/1.png"
    />
  </div>
  <div
    className="pokemonType"
  >
    Electric
  </div>
  <ul
    className="sc-dkzDqf bJxurm"
  >
    <li />
    <li />
    <li />
    <li />
  </ul>
</div>
`;
